src/yunohost/backup.py

```diff
@@ -49,111 +51,257 @@
-        # Check that output directory is empty
-        elif no_compress and os.listdir(output_directory):
-            raise MoulinetteError(errno.EIO,
-                m18n.n('backup_output_directory_not_empty'))
-
-        # Do not compress, so set temporary directory to output one and
-        # disable bind mounting to prevent data loss in case of a rm
-        # See: https://dev.yunohost.org/issues/298
-        if no_compress:
-            logger.debug('bind mounting will be disabled')
-            tmp_dir = output_directory
-            env_var['CAN_BIND'] = 0
-    else:
-        output_directory = ARCHIVES_PATH
+    Note: some future backups methods (like borg) are not able to specify a
+    different place than the original path. That's why the ynh_restore_file
+    helpers use primarily the SOURCE_PATH as argument.
 
-    # Create archives directory if it does not exists
-    if not os.path.isdir(ARCHIVES_PATH):
-        os.mkdir(ARCHIVES_PATH, 0750)
+    usage:
+        backup_manager = BackupManager(name="mybackup", description="bkp things")
+
+        # Add backup method to apply
+        backup_manager.add(BackupMethod.create('copy','/mnt/local_fs'))
+        backup_manager.add(BackupMethod.create('tar','/mnt/remote_fs'))
+
+        # Collect hooks and apps files
+        backup_manager.collect_files(["data"], ["wordpress"])
 
-    def _clean_tmp_dir(retcode=0):
-        ret = hook_callback('post_backup_create', args=[tmp_dir, retcode])
+        # Apply backup methods
+        backup_manager.backup()
+
+        # Clean tmp dir
+        backup_mananger.clean()
+
+    """
+
+    def __init__(self, name=None, description='', work_dir=None):
+        """
+        name: string The name of this backup (without spaces)
+        description: string A description for this future backup archive
+        work_dir: None|string A path where prepare the archive
+        """
+        self.description = description or ''
+        self.created_at = int(time.time())
+        self.apps_return = {}
+        self.hooks_return = {}
+        self.methods = []
+        self.paths_to_backup = []
+        # Define backup name
+        if not name:
+            name = self._define_backup_name()
+
+        self.name = name
+        self.work_dir = work_dir
+        if self.work_dir is None:
+            self.work_dir = os.path.join(BACKUP_PATH, 'tmp', name)
+            self.bindable = True
+        else:
+            self.bindable = False
+        self._init_work_dir()
+
+    @property
+    def info(self):
+        return {
+            'description': self.description,
+            'created_at': self.created_at,
+            'size': self.size,
+            'apps': self.apps_return,
+            'hooks': self.hooks_return
+        }
+
+    def __repr__(self):
+        return json.dumps(self.info)
+
+    def add(self, method):
+        self.methods.append(method)
+
+    def collect_files(self, hooks=[], apps=[]):
```

Comment:
Same remark here [about the "hook" terminology](https://github.com/YunoHost/yunohost/pull/275/files#r114081796)